### PR TITLE
Kirill.sitcevoy/computers

### PR DESCRIPTION
**Pull request by kievzenit**
Computers home task

### CODE REVIEW

**Review by horianyn — DISMISSED** [review]

File: CSharp_Net_module1_1_4_lab/CSharp_Net_module1_1_4_lab/Program.cs [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+            new Computer[] {desktop, laptop, server, server}
+            };
+
+            // preparing variables
+            int allComputersCount = 0;
+            int maxHDD = 0;
+            int minCPU = int.MaxValue;
+            int minMemory = int.MaxValue;
+            string computerWithLargestHDD = "";
+            string computerWithLowestProductivity = "";
```
Comment (Artya) [inline CSharp_Net_module1_1_4_lab/CSharp_Net_module1_1_4_lab/Program.cs]: please use one way of the variable declaring, here you have combined 2 ways with and without `var`

File: CSharp_Net_module1_1_4_lab/CSharp_Net_module1_1_4_lab/Program.cs [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+            server.HDD = 2048;
+
+            Computer[][] departments = new Computer[4][] {
+            new Computer[] {desktop, desktop, laptop, laptop, server},
+            new Computer[] {laptop, laptop, laptop},
+            new Computer[] {desktop, desktop, desktop, laptop, laptop},
+            new Computer[] {desktop, laptop, server, server}
+            };
+
+            // preparing variables
```
Comment (Artya) [inline CSharp_Net_module1_1_4_lab/CSharp_Net_module1_1_4_lab/Program.cs]: you don't need to leave such comments

File: CSharp_Net_module1_1_4_lab/CSharp_Net_module1_1_4_lab/Program.cs [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+                    // finding computer with lowest productivity
+                    if (departments[i][j].Cores < minCPU && departments[i][j].Memory < minMemory)
+                    {
+                        minCPU = departments[i][j].Cores;
+                        minMemory = departments[i][j].Memory;
+                        computerWithLowestProductivity = departments[i][j].ToString();
+                    }
+
+                    // changing memory to 8GB for all desktop computers
+                    if (departments[i][j].ToString() == ComputerType.Desktop.ToString())
```
Comment (Artya) [inline CSharp_Net_module1_1_4_lab/CSharp_Net_module1_1_4_lab/Program.cs]: Here you just guessing when you will have enum there and `ComputerType.Desktop`, i don't think this trick is good idea.

File: CSharp_Net_module1_1_4_lab/CSharp_Net_module1_1_4_lab/Program.cs [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+                    {
+                        departments[i][j].Memory = 8;
```
Comment (Artya) [inline CSharp_Net_module1_1_4_lab/CSharp_Net_module1_1_4_lab/Program.cs:94]: Are you sure that you will really change value in the array by this operation?

**Review by Artya — CHANGES_REQUESTED** [review]

File: CSharp_Net_module1_1_4_lab/CSharp_Net_module1_1_4_lab/Program.cs (hunk 1 above)
Comment (kievzenit) [reply CSharp_Net_module1_1_4_lab/CSharp_Net_module1_1_4_lab/Program.cs]: Fixed in commit f5e31f3a4612e2e146c57d592bbe6505c3d1dea4

File: CSharp_Net_module1_1_4_lab/CSharp_Net_module1_1_4_lab/Program.cs (hunk 2 above)
Comment (kievzenit) [reply CSharp_Net_module1_1_4_lab/CSharp_Net_module1_1_4_lab/Program.cs]: Fixed in commit f5e31f3a4612e2e146c57d592bbe6505c3d1dea4

File: CSharp_Net_module1_1_4_lab/CSharp_Net_module1_1_4_lab/Program.cs (hunk 3 above)
Comment (kievzenit) [reply CSharp_Net_module1_1_4_lab/CSharp_Net_module1_1_4_lab/Program.cs]: Fixed in commit 358be65e7ad42d38e026c83578e728f65e9c1657

File: CSharp_Net_module1_1_4_lab/CSharp_Net_module1_1_4_lab/Program.cs (hunk 4 above)
Comment (kievzenit) [reply CSharp_Net_module1_1_4_lab/CSharp_Net_module1_1_4_lab/Program.cs:94]: Yes, I'm totally sure, cause I'm not changing local copy of computer memory value, I'm changing value that in the heap

**Review by horianyn — APPROVED** [review]

**Review by SergiySimchuk — APPROVED** [review]
You missed count computers of every type, but I think you can do it easy) So, I Approve.

**Comment by kievzenit** [comment]
> You missed count computers of every type, but I think you can do it easy) So, I Approve.

Oh, really forgot

**Review by Artya — APPROVED** [review]

**Pull request merged**